scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -2133,19 +2133,21 @@ class FormatOps(
               def splits = Some(getSplitsForIf(ft, nft, t))
               def rightBrace = blockLast(t.thenp)
             })
-          case t: Term.For if !nft.right.is[T.KwDo] =>
-            // unsupported except for right brace
+          case Term.For(_, t: Term.Block)
```

Comment:
looks like we now have extra conditions. is that equivalent? keep in mind that right brace method may have been used not just for formatting but for rewrite, maybe end marker, and possibly something else. can we do a quick search on usage of `rightBrace` and see if there are cases we'd forgotten to cover with tests?